File: send-data-from-one-fragment-to-another-java-version/app/src/main/java/com/example/senddatafromonefragmmenttoanotherjavaversion/FirstFragment.java
package com.example.senddatafromonefragmmenttoanotherjavaversion;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

public class FirstFragment extends Fragment {

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        View view = inflater.inflate(R.layout.activity_first_fragment, container, false);

        Button btn = view.findViewById(R.id.btnSend);
        Communicator communicator = (Communicator) getActivity();

        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                EditText editText = view.findViewById(R.id.editText);
                String data = editText.getText().toString();
                assert communicator != null;
                communicator.passData(data);
            }
        });
        return view;
    }
}

File: send-data-from-one-fragment-to-another-fragment-using-kotlin/app/src/main/java/com/example/senddatafromonefragmenttoanotherusingkotlin/FirstFragment.kt
package com.example.senddatafromonefragmenttoanotherusingkotlin

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import kotlinx.android.synthetic.main.activity_first_fragment.view.*

class FirstFragment : Fragment() {
    private lateinit var communicator: Communicator

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val rootView = inflater.inflate(R.layout.activity_first_fragment, container, false) as ViewGroup
        communicator = activity as Communicator

        rootView.btnSend.setOnClickListener {
            communicator.passData(rootView.editText.text.toString())
        }
        return rootView
    }
}